analysis/selectQEDEvents.cpp

```diff
@@ -433,7 +453,18 @@ int main(int argc, char* argv[])
     eleSCEta_2   = electron2->GetEtaSC();
     eleSCPhi_2   = electron2->GetPhiSC();
     eleEoverP_2  = electron2->GetEoverP(); 
- 
+    ////////////////////////////////////////
+    nVtx = vertices.size();
+    for(auto vertex : vertices){
+     cout << "xVtx = " << vertex->GetPVertexX() << endl;
+     cout << "yVtx = " << vertex->GetPVertexY() << endl;
+     cout << "zVtx = " << vertex->GetPVertexZ() << endl;
+     xVtx = vertex->GetPVertexX();
+     yVtx = vertex->GetPVertexY();
+     zVtx = vertex->GetPVertexZ();
```

Comment:
I'm not sure I understand what this code does. It will overwrite xVtx value in each iteration, so you'll only keep the last vertex position - is that what you wanted?

Reply:
xVtz, yVtx, zVtx are initialized with -999 in this code and also in https://github.com/Pranati-Jana/LightByLight2018/blob/master/analysis/src/PhysObject.cpp#L52-#L54 . 
Maybe this can give -999 for events which are failing selectQED criteria but passing trigger criteria? 
@jniedzie  what do you think?

Reply:
Here I want to store the vertex info for the events which pass the dielectron selection criteria (if(electron1->GetCharge() == electron2->GetCharge()) continue;).

Reply:
Well, in the lines you pointed to they are initialized to -9999. But I agree that in general this -999 looks like an initial value in case they cannot be set properly from the ROOT tree. 

We discussed with @Pranati-Jana to check if some files are 100% good and some 100% bad - this could indicate that a fraction of input files just doesn't contain the vertex information for some reason. Otherwise, we should find a file where we have some -999 and analyze them closely.